src/core/include/openvino/core/graph_util.hpp

```diff
@@ -224,6 +224,7 @@ template <typename T>
 std::vector<std::shared_ptr<Node>> topological_sort(T root_nodes) {
     std::stack<Node*, std::vector<Node*>> nodes_to_do;
     std::unordered_set<Node*> nodes_done;
+    std::vector<std::tuple<Node* /*from*/, Node* /*to*/, int64_t /*index*/>> visited_edges;
```

Comment:
I'll work on moving it to std::unordered_set<Edge> where Edge is a custom hash-able struct

Reply:
Done